test/beginend-dired-test.el

```diff
@@ -33,80 +33,87 @@
+     ,@body))
 
 (describe "beginend in a dired buffer"
-  (describe "without dired-hide-details-mode"
-    (describe "without dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p))
-
-      (it "offers a command to go to last file"
-        (beginend-dired-test)))
-
-    (describe "with dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p 'omit))
-
-      (it "offers a command to go to last file with dired-omit-mode"
-        (beginend-dired-test 'omit))))
-
-  (describe "with-dired-hide-details-mode (also hides information lines)"
-    (describe "without dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p 'hide-details))
-
-      (it "offers a command to go to last file with dired-omit-mode"
-        (beginend-dired-test 'hide-details)))
-
-    (describe "with dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p 'omit 'hide-details))
-
-      (it "offers a command to go to last file with dired-omit-mode"
-        (beginend-dired-test 'omit 'hide-details))))
-
-  (describe "with-dired-hide-details-mode (does not hide information lines)"
-    (describe "without dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p 'hide-details 'keep-info-lines))
-
-      (it "offers a command to go to last file with dired-omit-mode"
-        (beginend-dired-test 'hide-details 'keep-info-lines)))
-
-    (describe "with dired-omit-mode"
-      (it "offers a command to go to first file"
-        (beginend-dired-test 'begin-p 'omit 'hide-details 'keep-info-lines))
-
-      (it "offers a command to go to last file with dired-omit-mode"
-        (beginend-dired-test 'omit 'hide-details 'keep-info-lines)))))
+  (it "ignores . and .. at the beginning"
+    (with-beginend-dired-test-buffer ("." ".." "dir1" "dir2")
+      (beginend-dired-test-begin "dir1")
+      (beginend-dired-test-end "dir2")))
+
+  (it "ignores . and .. at the end"
+    (with-beginend-dired-test-buffer ("dir1" "dir2" "." "..")
+      (beginend-dired-test-begin "dir1")
+      (beginend-dired-test-end "dir2")))
+
+  (it "ignores . at the beginning and .. at the end"
+    (with-beginend-dired-test-buffer ("." "dir1" "dir2" "..")
+      (beginend-dired-test-begin "dir1")
+      (beginend-dired-test-end "dir2")))
+
+  (it "ignores . and .. in the middle"
+    (with-beginend-dired-test-buffer ("dir1" "." ".." "dir2")
+      (beginend-dired-test-begin "dir1")
+      (beginend-dired-test-end "dir2")))
+
+  (it "ignores . and .. when they are hidden"
+    (with-beginend-dired-test-buffer ("dir1" "dir2")
+      (beginend-dired-test-begin "dir1")
+      (beginend-dired-test-end "dir2")))
+
+  (it "ignores . and .. when they are hidden and the directory is empty"
+    (with-beginend-dired-test-buffer ()
+      (beginend-dired-test-begin nil)
```

Comment:
should that be `point-min` here instead of `point-max`?

Reply:
Yes, it should be `point-min`.  Also, I replace `nil` with `bob` and `eob` which are more descriptive.